### PR TITLE
Feature/ajax delete song

### PR DESCRIPTION
**Pull request by goody153**
## Ticket : 
  - [ajax delete song](https://trello.com/c/GCdtR3pM)

## Summary: 
 - Removes song from the template
 - Show description on playlists

### CODE REVIEW

File: assets/js/playlist.js [hunk 1]
```diff
@@ -36,4 +35,17 @@
...
       }
     });
-  });
\ No newline at end of file
+  });
+
+  // ajax for deleting songs
+  $(document).on('click', '.deleteSong', function( event ){
+    $.ajax({
+      method: 'GET',
```
Comment (mcandidier) [inline assets/js/playlist.js]: delete function should be used POST method instead of GET

Comment (goody153) [reply assets/js/playlist.js]: Updated.

File: assets/js/playlist.js [hunk 2]
```diff
@@ -59,6 +62,37 @@
...
+    }).done(function(response){
+      //remove song from the template
+      $("#"+response.song_id).parent( "li" ).remove()
+      song_ids.pop();
+      console.log(song_ids);
+    });
+  });
+
+  // function for creating csrf_token 
+  function getCookie(name) {
```
Comment (mcandidier) [inline assets/js/playlist.js]: where do you get this? you already pass csrf_token from your form, check your form serializer.

Comment (mcandidier) [reply assets/js/playlist.js]: remove this getCookie function, this is not the proper way to do this.

File: assets/js/playlist.js [hunk 3]
```diff
@@ -10,10 +10,15 @@
       url: $(this).attr('action'),
       data: data
     }).done(function(response){
+      // add videoID to player
+      song_ids.push(response.link);
+      // create the necessary csrftoken for the delete song form
+      var csrftoken = getCookie('csrftoken');
       // songEntry will be used to append on the songlist
-      songEntry = '<div class="media">'
+      songEntry = '<li id="song-state-'+ ( song_ids.length ) +'" class="">'
```
Comment (mcandidier) [inline assets/js/playlist.js]: dont make it as a global variable.

File: assets/js/playlist.js [hunk 4]
```diff
@@ -10,10 +10,15 @@
       url: $(this).attr('action'),
       data: data
     }).done(function(response){
+      // add videoID to player
+      song_ids.push(response.link);
+      // create the necessary csrftoken for the delete song form
+      var csrftoken = getCookie('csrftoken');
```
Comment (mcandidier) [inline assets/js/playlist.js]: remove this, csrf_token, token already pass from your form.

File: assets/js/playlist.js [hunk 5]
```diff
@@ -59,6 +62,37 @@
...
+    event.preventDefault();
+    $.ajax({
+      method: 'POST',
+      url: $(this).attr('action'),
+      data: $(this).serialize()
+    }).done(function(response){
+      //remove song from the template
+      $("#"+response.song_id).parent( "li" ).remove()
+      song_ids.pop();
+      console.log(song_ids);
```
Comment (mcandidier) [inline assets/js/playlist.js]: remove console log. make sure to remove it everytime you commit.

File: assets/js/playlist.js (hunk 2 above)
Comment (goody153) [reply assets/js/playlist.js]: I will change it, sir. I found that here 
https://docs.djangoproject.com/en/1.11/ref/csrf/

File: assets/js/playlist.js (hunk 5 above)
Comment (goody153) [reply assets/js/playlist.js]: Updated.

File: assets/js/playlist.js (hunk 3 above)
Comment (goody153) [reply assets/js/playlist.js]: Updated.

File: assets/js/playlist.js (hunk 4 above)
Comment (goody153) [reply assets/js/playlist.js]: Updated.

**Pull request merged**